### PR TITLE
Feature/us6263 form fields

### PR DESCRIPTION
**Pull request by zarahzachz**
Given a designer/developer using the DDK
When I need to reference styles for the inline giving tool 
Then I should find them in the Digital Design Kit

Add form elements to Atoms section of DDK

Components in use on inline giving include:
Text input
Select/drop down
Radio
Checkbox
Alerts (use new styles from Rob on these, this will change inline giving - discuss w/ Brian as needed)
Examples (placeholder text)

Include treatment for both dark & light themes

Form elements should include validation styles
See Rob's list of variants that should be included for each bucket https://docs.google.com/a/helloample.com/spreadsheets/d/1SPvEI6cjEv6t87mHTnSHOcj2g5TR_weM1l_GVaBwUGY/edit?usp=sharing

### CODE REVIEW

File: assets/stylesheets/_overrides.scss [hunk 1]
```diff
@@ -2,61 +2,64 @@
...
+  $brand-primary: $cr-blue !default;
+  $brand-success: $cr-green !default;
+  $brand-info: $cr-cyan-lighter !default;
+  $brand-warning: $cr-orange !default;
+  $brand-danger: $cr-magenta !default;
+  $gray-darker: $cr-gray-darker !default;
+  $gray-dark: $cr-gray-dark !default;
+  $gray: $cr-gray !default;
+  $gray-light: $cr-gray-light !default;
+  $gray-lighter: $cr-gray-lighter !default;
```
Comment (tcmacdonald) [inline assets/stylesheets/_overrides.scss]: Why are we adding whitespace here?

File: assets/stylesheets/_variables.scss [hunk 2]
```diff
@@ -7,17 +7,20 @@ $base-font-size: 16px;
 
 // Colors
 $cr-gray: rgba(151,151,151,1);           // #979797
+$cr-gray-alpha: rgba(151,151,151,.25);   // #979797
 $cr-gray-dark: rgba(115,115,115,1);      // #737373
 $cr-gray-darker: rgba(77,77,77,1);       // #4D4D4D
 $cr-gray-light: rgba(205,205,205,1);     // #CDCDCD
 $cr-gray-lighter: rgba(231,231,231,1);   // #E7E7E7
+$cr-gray-lightest: rgba(244,244,244,1);  // #F4F4F4
```
Comment (tcmacdonald) [inline assets/stylesheets/_variables.scss]: This convention deviates from Bootstrap's [predefined colors](http://getbootstrap.com/customize/#colors). I'd prefer to use the `lighten()` method in places where you need a slightly lighter color of `$gray-light` just to prevent any confusion.

File: assets/stylesheets/base/_global.scss [hunk 3]
```diff
@@ -2,7 +2,28 @@
...
+:focus {
+  outline: 0;
+}
+
 body {
   font-size: $base-font-size;
   @extend .font-family-base;
+}
+
+%clearfix {
```
Comment (tcmacdonald) [inline assets/stylesheets/base/_global.scss]: This mixin should already be defined by Bootstrap. 

https://github.com/twbs/bootstrap-sass/blob/master/assets/stylesheets/bootstrap/mixins/_clearfix.scss

File: assets/stylesheets/components/_alerts.scss [hunk 4]
```diff
@@ -0,0 +1,27 @@
+.alert {
+  border-radius: 0;
+  line-height: 1.125rem;
+
+
+  &.alert-success {
+    background: lighten($brand-success, 40);
```
Comment (tcmacdonald) [inline assets/stylesheets/components/_alerts.scss]: Can we refactor this to use Bootstrap's `alert-variant()` mixin?

https://github.com/twbs/bootstrap-sass/blob/master/assets/stylesheets/bootstrap/mixins/_alerts.scss

File: assets/stylesheets/components/_form-validation.scss [hunk 5]
```diff
@@ -0,0 +1,29 @@
+.form-group {
+  &.has-success,
+  &.has-error {
```
Comment (tcmacdonald) [inline assets/stylesheets/components/_form-validation.scss]: Bootstrap also provides the `.has-warning` selector. Have we considered overriding styles for that state, in addition to error and success?

File: assets/stylesheets/components/_forms.scss [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+    border: $input-border;
+    border-radius: $input-border-radius;
+    color: $input-group-addon-border-color;
+    font-size: 1.125rem;
+  }
+}
+
+// Datepicker
+
+datepicker {
```
Comment (tcmacdonald) [inline assets/stylesheets/components/_forms.scss]: Is datepicker actually being introduced in this PR? Thought we split that out as a separate story...

File: assets/stylesheets/components/_forms.scss [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+            border-bottom-left-radius: 0;
+            border-top-right-radius: 4px;
+            border-bottom-right-radius: 4px;
+          }
+        }
+      }
+    }
+  }
+
+  .btn {
```
Comment (tcmacdonald) [inline assets/stylesheets/components/_forms.scss]: This looks redundant to styles introduced in a previous commit... 

https://github.com/crdschurch/crds-styles/blob/development/assets/stylesheets/components/_buttons.scss

Comment (tcmacdonald) [inline assets/stylesheets/components/_forms.scss]: Ah – I see below that you've killed `_buttons.scss` and move those styles into here. Can you talk to the benefit you in doing that? Is that how Bootstrap has architected its styles too?

File: assets/stylesheets/themes/_dark.scss [hunk 8]
```diff
@@ -14,4 +14,63 @@
   h2 {
     color: $headings-color;
   }
+
+  // Datepicker
+
+  datepicker {
```
Comment (tcmacdonald) [inline assets/stylesheets/themes/_dark.scss]: Similar to my comment above... this doesn't seem relevant to this PR.

File: assets/stylesheets/components/_headings.scss [hunk 9]
```diff
@@ -0,0 +1,42 @@
+h1 {
```
Comment (tcmacdonald) [inline assets/stylesheets/components/_headings.scss]: These elements have been styles elsewhere... 

https://github.com/crdschurch/crds-styles/blob/feature/US6263-form-fields/assets/stylesheets/components/_typography.scss#L36-L92

If you intended to extract these selectors from `_typography.scss` – I think you forgot to do so.

**Comment by zarahzachz** [comment]
I've made changes to the styles. Please review. 

This branch corresponds with crds-styleguide: `feature/US6263-form-fields`

File: assets/stylesheets/components/_form-validation.scss [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  &.has-success {
+    border-right-color: $brand-success;
+  }
+
+  &.has-error {
+    border-right-color: $brand-danger;
+
+    .helper-block {
```
Comment (tcmacdonald) [inline assets/stylesheets/components/_form-validation.scss]: This should be `.help-block`
